### PR TITLE
PLANET-5323 Added JS acceptance tests for Submenu block

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**
Ref. [PLANET 5353](https://jira.greenpeace.org/browse/PLANET-5323)

image

To test, run `npm install` and `npm run test`.
Use `npm run test:watch` to run the tests in slow motion and watch execution in browser.

### CODE REVIEW

File: tests/js/submenu.spec.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    await expect(page).toMatchElement('.submenu-block h2', { text: 'Submenu title' });
+
+    // Check the submenu links for 3 levels and their corresponding list styles.
+    await expect(page).toMatchElement('li.list-style-none a[href=\\#h2-no-1]', { text: 'H2 no 1' });
+    await expect(page).toMatchElement('li.list-style-bullet a[href=\\#h2\\.h3-no-2]', { text: 'H2.H3 no 2' });
+    await expect(page).toMatchElement('li.list-style-number a[href=\\#h2\\.h3\\.h4-no-1]', { text: 'H2.H3.H4 no 1' });
+
+    // On click of submenu link, Move focus to actual header text.
+    await clickElementByText( 'a', 'H2.H3 no 1' );
+    const hieghtFromTop = await page.evaluate(
```
Comment (lithrel) [inline tests/js/submenu.spec.js:159]: hieght -> h**ei**ght

**Review by lithrel — APPROVED** [review]
Looks good to me 👍 
Couldn't test it, it keeps running in timeout locally :/

**Comment by sagarsdeshmukh** [comment]
> Looks good to me 👍
> Couldn't test it, it keeps running in timeout locally :/

Isn't it work with the `npm run test:watch` cmd ?
or may be try the custom timeout value, like - `npm run env JEST_TIMEOUT=40000 WP_USERNAME=admin WP_PASSWORD=admin WP_BASE_URL=https://www.planet4.test wp-scripts test-e2e`

**Comment by lithrel** [comment]
I tried with a 60000 timeout delay :) I can't `watch` from inside the container (cannot open display), and `install` fails on my machine outside of the container, sooo, I have stuff to resolve on my machine before I can run this ^^

**Comment by sagarsdeshmukh** [comment]
Added a separate test for [submenu frontend](tests/js/submenu.frontend.spec.js) -
image

@lithrel - The reason of failing test on your local dev env will be the blocks repo branch, I tested the JS tests against `planet-4811-use-get-blocks-in-editor` branch. AFAIK it has the latest wysiwyg submenu block code. Sorry, I forgot to mention it in test steps.

**Pull request merged**